shapely/set_operations.py

```diff
@@ -501,3 +503,110 @@ def coverage_union_all(geometries, axis=None, **kwargs):
     # create_collection acts on the inner axis
     collections = lib.create_collection(geometries, GeometryType.GEOMETRYCOLLECTION)
     return lib.coverage_union(collections, **kwargs)
+
+
+@multithreading_enabled
```

Comment:
```suggestion
@requires_geos("3.12.0")
@multithreading_enabled
```

(and same for the `_all` version below)